pkg/instrumentation/annotation.go

```diff
@@ -23,15 +23,16 @@ import (
 const (
 	// annotationInjectJava indicates whether java auto-instrumentation should be injected or not.
 	// Possible values are "true", "false" or "<Instrumentation>" name.
-	annotationInjectJava = "instrumentation.opentelemetry.io/inject-java"
```

Comment:
can you explain the intentions for changing this annotation?

Reply:
Yeah I'm happy to split out the annotation change if it makes sense.

I found our current, SDK-agnostic (?) sdk.go uses this Java annotation. I need it to be able to support multiple languages and came up with this change. Any idea that could work better?

Reply:
Could you please explain why we need two annotations? 

We can have `"instrumentation.opentelemetry.io/inject-java"`, `"instrumentation.opentelemetry.io/inject-node"` and then `"instrumentation.opentelemetry.io/inject" - to just inject the basic configuration (users could use the operator as a control plane to manage instrumentations).

Reply:
I don't think we want to allow multiple languages on the same pod as that shouldn't ever be a use case I think. While we could check the annotations and fail if there are multiple this split seemed to model more directly that only one language can be injected.

I made language required without thinking too much but could definitely remove that to allow the control plane only injection if this otherwise makes sense.

Reply:
I don't see much value of limiting it to only a single language, quite the opposite. 

What I don't like is to have two annotations from the UX perspective. 

```
	annotationInject   = "instrumentation.opentelemetry.io/inject"
	annotationLanguage = "instrumentation.opentelemetry.io/language"
```

Reply:
I believe these sort of annotations are always copy-pasted or frameworked, so I don't think there is a usability difference between the two patterns. If we think there is a use case for multiple annotations, then it makes sense but for example having both `otel.inject=true` and `otel.inject-java=true` on the same pod would be redundant and I think that leads to some cognitive dissonance.

Great to hear everyone's ideas on the matter :)

Reply:
Discussed with @pavolloffay offline and for now sticking to the current annotation format, but we'll need to address it soon. We can see the awkwardness here

https://github.com/open-telemetry/opentelemetry-operator/pull/507/files#diff-92f790ce871022e5e7de195a2f96235dac6ac3fb04337974f8584c7c020a67cbR48

where common config can be injected from different instrumentations in a surprising way. Having separate annotations for the instrumentation config and what languages to use in the future will hopefully solve that.

Reply:
+1 let's discuss it in a separate issue. maybe there are other ways we can mitigate the issue